runtime/lua/vim/lsp/util.lua

```diff
@@ -1437,6 +1437,9 @@ local function make_position_param()
   local row, col = unpack(api.nvim_win_get_cursor(0))
   row = row - 1
   local line = api.nvim_buf_get_lines(0, row, row+1, true)[1]
+  if not line then
```

Comment:
how can the _current_ buffer be unloaded ...? And why wouldn't we just load it instead? e.g.

```
if not api.nvim_buf_is_loaded(0) then
  vim.fn.bufload('.')
end
```

@gysddn can you try that, and see if it fixes the issue for you? Though I would guess not, since, as I mentioned, current buffer must always be loaded...

Reply:
Ah I read https://github.com/neovim/neovim/issues/12623#issuecomment-675360894 , that makes sense, it happens during startup. (helps to mention this in the commit message)

Sure enough, `source_startup_scripts` is done before `edit_buffers`:

https://github.com/neovim/neovim/blob/9bb552875d205d2f869c66137563f93b77a6d08e/src/nvim/main.c#L362

https://github.com/neovim/neovim/blob/9bb552875d205d2f869c66137563f93b77a6d08e/src/nvim/main.c#L480

Reply:
Yeah, I think there is a logic issue in the script, or at least calling it in the startup file doesn't seem right. First I thought maybe informing the user when buffer isn't loaded in nvim_buf_get_lines() with an error would help but then I saw they have a defined behaviour already, so I went with this instead.

I'll try to be more thorough with my commit messages next time :hand: